examples/04_training/06_train_icemix_model.py

```diff
@@ -0,0 +1,277 @@
+"""Example of training Model.
+
+This example is based on Icemix solution proposed in
+https://github.com/DrHB/icecube-2nd-place.git (2nd place solution). There are a
+few things to note about this example:
+"""
+
+import os
+from typing import Any, Dict, List, Optional
+
+from pytorch_lightning.loggers import WandbLogger
+from torch.optim.adam import Adam
+from torch.optim.lr_scheduler import ReduceLROnPlateau
+
+from graphnet.constants import EXAMPLE_DATA_DIR, EXAMPLE_OUTPUT_DIR
+from graphnet.data.constants import FEATURES, TRUTH
+from graphnet.models import StandardModel
+from graphnet.models.detector.prometheus import Prometheus
+from graphnet.models.gnn import DeepIce
+from graphnet.models.graphs import KNNGraph
+from graphnet.models.graphs.nodes import IceMixNodes
+from graphnet.models.task.reconstruction import (
+    DirectionReconstructionWithKappa,
+)
+from graphnet.training.labels import Direction
+from graphnet.training.loss_functions import VonMisesFisher3DLoss
+from graphnet.training.utils import make_train_validation_dataloader
+from graphnet.utilities.argparse import ArgumentParser
+from graphnet.utilities.logging import Logger
+
+# Constants
+features = FEATURES.PROMETHEUS
+truth = TRUTH.PROMETHEUS
+
+
+def main(
+    path: str,
+    pulsemap: str,
+    target: str,
+    truth_table: str,
+    gpus: Optional[List[int]],
+    max_epochs: int,
+    early_stopping_patience: int,
+    batch_size: int,
+    num_workers: int,
+    wandb: bool = False,
+) -> None:
+    """Run example."""
+    # Construct Logger
+    logger = Logger()
+
+    # Initialise Weights & Biases (W&B) run
+    if wandb:
+        # Make sure W&B output directory exists
+        wandb_dir = "./wandb/"
+        os.makedirs(wandb_dir, exist_ok=True)
+        wandb_logger = WandbLogger(
+            project="example-script",
+            entity="graphnet-team",
+            save_dir=wandb_dir,
+            log_model=True,
+        )
+
+    logger.info(f"features: {features}")
+    logger.info(f"truth: {truth}")
+
+    # Configuration
+    config: Dict[str, Any] = {
+        "path": path,
+        "pulsemap": pulsemap,
+        "batch_size": batch_size,
+        "num_workers": num_workers,
+        "target": target,
+        "early_stopping_patience": early_stopping_patience,
+        "fit": {
+            "gpus": gpus,
+            "max_epochs": max_epochs,
+            "distribution_strategy": "ddp_find_unused_parameters_true",
+        },
+    }
+
+    graph_definition = KNNGraph(
+        detector=Prometheus(),
+        node_definition=IceMixNodes(
+            input_feature_names=features,
+            max_pulses=128,
+            z_name="sensor_pos_z",
+            hlc_name=None,
+            add_ice_properties=True,
```

Comment:
`prometheus` is a reference to  an open-source simulation of `ORCA150`, which is a water-based detector. So I think we should set `add_ice_properties = False` here, since you're training from scratch.